tests/test_agents/base.py

```diff
@@ -434,29 +420,43 @@ class BaseTestEnd2EndAgentCatchup(BaseTestEnd2End):
       node can receive the responses
     """
 
-    # mandatory argument
-    stop_string: str
-
-    restart_after: int = 60
-    wait_before_stop: int = 15
-
-    def setup(self) -> None:
-        """Set up the test."""
-        if not hasattr(self, "stop_string"):
-            pytest.fail("'stop_string' is a mandatory argument.")
+    nb_nodes: int = 0  # number of agents with tendermint nodes
+
+    # configuration specific for agent restart
+    stop_string: str  # mandatory argument if n_terminal > 0
+    n_terminal: int = 0  # number of agents to be restarted
+    wait_to_kill: int = 0  # delay the termination event
+    restart_after: int = 60  # how long to wait before restart
+    wait_before_stop: int = 15  # how long to check logs
+
+    def check(self) -> None:
+        """Check pre-conditions of the test"""
+        if self.n_terminal > self.nb_nodes:
+            fail_msg = "Cannot terminate {nb_nodes} out of {n_terminal} agents:"
+            pytest.fail(
+                fail_msg.format(nb_nodes=self.nb_nodes, n_terminal=self.n_terminal)
+            )
```

Comment:
We should instead check that `n_terminal` is not bigger than the faulty threshold.